src/tools/kata-ctl/src/check.rs

```diff
@@ -89,13 +95,49 @@ pub fn run_network_checks() -> Result<()> {
     Ok(())
 }
 
+fn get_kata_version_by_url(url: &str) -> Result<String, reqwest::Error> {
+    let content = reqwest::blocking::Client::new()
+        .get(url)
+        .header(CONTENT_TYPE, "application/json")
+        .header(USER_AGENT, "kata")
+        .send()?
+        .json::<HashMap<String, Value>>()?;
+
+    let version = content["tag_name"].as_str().unwrap();
+    Ok(version.to_string())
+}
+
+fn handler(a: reqwest::Error) {
+    if a.is_connect() {
+        println!("http connection failure: connection refused");
+    }
+
+    if a.is_timeout() {
+        println!("http connection failure: connection timeout");
+    }
+
+    if a.is_builder() {
+        println!("http connection failure: url malformed");
+    }
+
+    if a.is_decode() {
+        println!("http connection failure: unable to decode response body");
+    }
+}
+
 pub fn check_version() -> Result<()> {
+    match get_kata_version_by_url(KATA_GITHUB_URL) {
+        Ok(version) => println!("Version: {}", version),
+        Err(e) => handler(e),
+    }
+
     Ok(())
 }
```

Comment:
You could "simplify" this slightly (in terms of numbers of lines, atleast ;) like this:

```suggestion
pub fn check_version() -> Result<()> {
    let version = get_kata_version_by_url(KATA_GITHUB_URL).map_err(|e| handle_reqwest_error(e))?;

    println!("Version: {}", version);

    Ok(())
}
```

Reply:
@jodh-intel one more, thanks